components/application-registry/internal/metadata/specification/minio/service.go

```diff
@@ -75,24 +86,55 @@ func (s *service) Get(id string) ([]byte, []byte, []byte, apperrors.AppError) {
 }
 
 func (s *service) Remove(id string) apperrors.AppError {
-	apperr := s.repository.Remove(bucketName, makeFilePath(id, documentationFileName))
+	apperr := s.tryRemoveWithDifferentExtensions(id, documentationFileName)
 	if apperr != nil {
 		return apperr
 	}
 
-	apperr = s.repository.Remove(bucketName, makeFilePath(id, apiSpecFileName))
+	apperr = s.tryRemoveWithDifferentExtensions(id, apiSpecFileName)
 	if apperr != nil {
 		return apperr
 	}
 
-	apperr = s.repository.Remove(bucketName, makeFilePath(id, eventsSpecFileName))
+	apperr = s.tryRemoveWithDifferentExtensions(id, eventsSpecFileName)
 	if apperr != nil {
 		return apperr
 	}
 
 	return nil
 }
 
+func (s *service) tryGetWithDifferentExtensions(id, fileName string) ([]byte, apperrors.AppError) {
+	var apperr apperrors.AppError
+	for _, ext := range extensions {
+		data, apperror := s.repository.Get(bucketName, makeFilePath(id, fmt.Sprintf("%s%s", fileName, ext)))
+		if apperror == nil {
+			return data, nil
+		}
+		if apperr != nil {
+			apperr.Append(apperror.Error(), apperr)
+		} else {
+			apperr = apperror
+		}
+	}
+	return nil, apperr
+}
+
+func (s *service) tryRemoveWithDifferentExtensions(id, fileName string) apperrors.AppError {
+	var apperr apperrors.AppError
+	for _, ext := range extensions {
```

Comment:
I would argue whether we need a loop here. Maybe implementing it as two function calls would make it more readable. Just a suggestion, we should discuss that.

Reply:
After discussion we assume that the readability won't change much and this solution provides easier future modifications nad extendability